ignite/metrics/frequency.py

```diff
@@ -58,21 +68,9 @@ def update(self, output: int) -> None:
 
     @sync_all_reduce("_n", "_elapsed")
     def compute(self) -> float:
-        time_divisor = 1.0
-
-        if idist.get_world_size() > 1:
-            time_divisor *= idist.get_world_size()
-
         # Returns the average processed objects per second across all workers
-        return self._n / self._elapsed * time_divisor
+        return int(self._n / self._elapsed * idist.get_world_size())
 
-    def completed(self, engine: Engine, name: str) -> None:
-        engine.state.metrics[name] = int(self.compute())
-
-    # TODO: see issue https://github.com/pytorch/ignite/issues/1405
-    def attach(  # type: ignore
-        self, engine: Engine, name: str, event_name: Events = Events.ITERATION_COMPLETED
-    ) -> None:
-        engine.add_event_handler(Events.EPOCH_STARTED, self.started)
-        engine.add_event_handler(Events.ITERATION_COMPLETED, self.iteration_completed)
-        engine.add_event_handler(event_name, self.completed, name)
+    # override the method attach() of Metrics to define a different default value for usage
+    def attach(self, engine: Engine, name: str, usage: Union[str, MetricUsage] = FrequencyWise()) -> None:
```

Comment:
Maybe, we can try to use [typing overload](https://python.readthedocs.io/en/stable/library/typing.html#typing.overload) to keep BC with the previous signature ?

Reply:
I took a look but it doesn't seem simple to use typing overload because argument names are different (to fit the `Metrics.attach()` method).